server/routes/api.js

```diff
@@ -67,13 +68,21 @@ router.get('/messages', isLoggedIn, function(req, res, next) {
     initData.limit = parseInt(defaultLimit, 10);
   }
   if (pdwMode) {
-    var subquery = db.from('capcodes').where('ignore', '=', 0).select('id')
+    if (adminShow && req.isAuthenticated()) {
+      var subquery = db.from('capcodes').where('ignore', '=', 1).select('id')
+    } else {
+      var subquery = db.from('capcodes').where('ignore', '=', 0).select('id')
+    }    
   } else {
     var subquery = db.from('capcodes').where('ignore', '=', 1).select('id')
   }
   db.from('messages').where(function () {
     if (pdwMode) {
-      this.from('messages').where('alias_id', 'in', subquery)
+      if (adminShow && req.isAuthenticated()) {
```

Comment:
Same thing here, if you move the check up 1 line you don't have to double up the query 😁